src/ImageSharp/Processing/Processors/Dithering/OrderedDither.cs

```diff
@@ -145,24 +144,25 @@ public OrderedDither(uint length)
                 in ditherOperation);
         }
 
-        [MethodImpl(InliningOptions.ShortMethod)]
+        // Spread assumes an even colorspace distribution and precision.
+        // Cubed root used because we always compare to Rgb.
+        // https://bisqwit.iki.fi/story/howto/dither/jy/
+        // https://en.wikipedia.org/wiki/Ordered_dithering#Algorithm
+        internal static int CalculatePaletteSpread(int colors) => (int)(255 / (Math.Pow(colors, 1.0 / 3) - 1));
```

Comment:
This will still give you bad results with some palettes.  It's not really a question of how many color channels are in the image but rather how many color channels are represented in the palette.  Consider the following 4-step greyscale palette

```
new[] { 0x000000u, 0x555555u, 0xAAAAAAu, 0xFFFFFFu };
```

Here the spread should be 85 (0x55), which is `255/(4 - 1)`.  The above calc will give you `255 / (cbrt(4) - 1)`, which comes out to 434 instead and will lead to overblown contrast again.  A black and white palette would give you a spread of 981 instead of the correct 255.

What you want to do here is build a check like `palette.IsGreyscale` and only take the cube root if that's false.  And just to be absolutely safe, I'd clamp the value of the denominator to a min of 1 (or clamp the result to a max of 255).  You could still be given an RGB palette that's less than 8 colors.

Reply:
I have been wondering about this tbh. We currently have no mechanism in place to determine whether or not the palette is greyscale. 

We also use the same operations to handle automatically resolved palettes from quantization so whatever we did we would have to be able to provide that option there also.

Reply:
Left a TODO for now. But thinking about some sort of enhancement for `PixelTypeInfo`